syntax = "proto3";

package ulens;

service Illuminate {
	rpc Illum (IllumReq) returns (SimpleReply) {}
	rpc Clear (SimpleReq) returns (SimpleReply) {}
	rpc Get (SimpleReq) returns (ImageReply) {}
}

message IllumReq{
	float x = 1; 
	float y = 2; 
	float z = 3; 
	float c = 4; 
}
message SimpleReq{
	string msg = 1; 
}
message SimpleReply{
	string msg = 1; 
}
message ImageReply{
	int32 w = 1;
	int32 h = 2; 
	bytes data = 3;
}
